pkgs/development/tools/xcbuild/developer.nix

```diff
@@ -0,0 +1,142 @@
+    Name = "Yacc";
+    Description = "Yacc Parser Generator";
+    InputFileTypes = ["sourcecode.yacc"];
+    ExecPath = "${yacc}/bin/yacc";
+    BuiltinJambaseRuleName = "ProcessUsingYacc";
+    ExecDescription = "Yacc $(InputFile)";
+    ProgressDescription = "Yaccing $(CommandProgressByType) files";
+    IsArchitectureNeutral = "YES";
+    Options = [
+	{
+	    Name = "YACC_GENERATED_FILE_STEM";
+	    Type = "enum";
+	    Values = ["Standard" "InputFileStem"];
+	    DefaultValue = "Standard";
+	    Category = "BuildOptions";
+	}
+	{
+	    Name = "YACC_INSERT_LINE_DIRECTIVES";
+	    Type = "bool";
+	    DefaultValue = "YES";
+	    Category = "BuildOptions";
+	    CommandLineArgs = { YES = []; NO = ["-l"]; };
+	}
+	{
+	    Name = "YACC_GENERATE_DEBUGGING_DIRECTIVES";
+	    Type = "bool";
+	    DefaultValue = "NO";
+	    Category = "BuildOptions";
+	    CommandLineArgs = { YES = ["-t"]; NO = []; };
+	}
+	{
+	    Name = "YACCFLAGS";
+	    Type = "StringList";
+	    DefaultValue = "";
+	    FileTypes = ["sourcecode.yacc"];
+	    Category = "BuildOptions";
+	}
+    ];
+  };
+
+  Lex = {
+    Identifier = "com.apple.compilers.lex";
+    Type = "Tool";
+    Name = "Lex";
+    Class = "PBXCompilerSpecificationLex";
+    Description = "Lex Scanner Generator";
+    ExecPath = "${flex}/bin/flex -X";
+    ExecDescription = "Lex $(InputFile)";
+    ProgressDescription = "Lexing $(CommandProgressByType) files";
+    InputFileTypes = ["sourcecode.lex"];
+    BuiltinJambaseRuleName = "ProcessUsingLex";
+    IsArchitectureNeutral = "YES";
+    Options = [
+	{
+	    Name = "LEX_CASE_INSENSITIVE_SCANNER";
+	    Type = "Boolean";
+	    DefaultValue = "NO";
+	    Category = "BuildOptions";
+	    CommandLineFlag = "-i";
+	}
+	{
+	    Name = "LEX_SUPPRESS_DEFAULT_RULE";
+	    Type = "Boolean";
+	    DefaultValue = "NO";
+	    Category = "BuildOptions";
+	    CommandLineFlag = "-s";
+	}
+	{   Name = "LEX_INSERT_LINE_DIRECTIVES";
+	    Type = "Boolean";
+	    DefaultValue = "YES";
+	    Category = "BuildOptions";
+	    CommandLineArgs = {
+		NO = ["-L"];
+	    };
+	}
+	{
+	    Name = "LEXFLAGS";
+	    Type = "StringList";
+	    DefaultValue = "";
+	    FileTypes = [
+		"sourcecode.lex"
+	    ];
+	    Category = "BuildOptions";
+	}
+	{
+	    Name = "LEX_SUPPRESS_WARNINGS";
+	    Type = "Boolean";
+	    DefaultValue = "NO";
+	    Category = "Warnings";
+	    CommandLineFlag = "-w";
+	}
+    ];
+  };
+
+  # others needed:
+  ## Cpp
+  ## Dtrace
+  ## Iconutil
+  ## LSRegisterURL
+  ## Ld
+  ## Libtool
+  ## Lipo
+  ## MiG
+  ## OSACompile
+  ## OpenCL
+  ## PBXCp
+  ## ProductTypeValidationTool
+  ## ResMerge
+  ## Rez
+  ## StripSymbols
+  ## TAPI
+  ## TiffUtil
+  ## UnifDef
+in
+
+stdenv.mkDerivation {
+  name = "Xcode.app";
+  buildInputs = [ xcbuild ];
+  buildCommand = ''
+    mkdir -p $out/Contents/Developer/Library/Xcode/Specifications/
+    cp ${xcbuild}/Library/Xcode/Specifications/* $out/Contents/Developer/Library/Xcode/Specifications/
```

Comment:
They aren't super well tested yet since the set isn't complete, but the hope is that they'd be for exactly this. (If you use xcbuild's CMake install, it can setup the directory structure for you, but not sure if that's common to use with Nix.)